staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -255,7 +255,7 @@ func (j *JSONPath) evalArray(input []reflect.Value, node *ArrayNode) ([]reflect.
 			params[1].Value = value.Len()
 		}
 
-		if params[1].Value < 0 {
+		if params[1].Value <= 0 {
```

Comment:
a few questions:
* how does changing the condition to include `0` relate to a negative index?
* does the same issue exist on line 251?
* we need a test demonstrating the bug without this change, and demonstrating this fixes it

Reply:
@liggitt  

- The data stores in params indicate a slice of array in the form of start-end-step. https://github.com/kubernetes/kubernetes/blob/d857790d360c2f9f86a47964017c08e409312551/staging/src/k8s.io/client-go/util/jsonpath/node.go#L140

- Here what we trying to do is dealing with negative index of array, container[-1] means the last container, container[-2] means the second to last....

- For example, when user want to refer container[-2], the data in params would be [-2, -1, 0]. The condition in L251 reverts param[0] to len-2. The condition in L258 reverts param[0] to len-1. So it becomes [len-2, len-1, 0], which equals container[len-2]

- When it comes to container [-1]. the data in params would be [-1, 0, 0]. You can see that the current condition would not revert param[1] which cause the bug. So we should only change the condition in L251.

Reply:
I will add a test to confirm this